### PR TITLE
SK-1545: multiple credentials

### PR DESCRIPTION
**Pull request by lukeg90**
SK-1545

### CODE REVIEW

File: src/components/ServiceEndpoints/ServiceEndpoint.tsx [hunk 1]
```diff
@@ -1,54 +1,93 @@
...
+  endpointType: string,
+): json is KiltPublishedCredentialCollectionV1 {
+  if (endpointType !== KiltPublishedCredentialCollectionV1Type) {
+    return false;
+  }
+  if (!Array.isArray(json)) {
+    return false;
+  }
+  const [{ credential }] = json as KiltPublishedCredentialCollectionV1;
+  return Credential.isICredential(credential);
```
Comment (arty-name) [inline src/components/ServiceEndpoints/ServiceEndpoint.tsx:40]: Should this check other array items as well?

File: src/components/ServiceEndpoints/ServiceEndpoint.tsx [hunk 2]
```diff
@@ -1,54 +1,93 @@
...
-      const result = await response.json();
-      setCredential(result);
+      if (Credential.isICredential(json)) {
+        setCredentials([json]);
+        return;
+      }
 
-      if (!did) {
-        throw new Error('No DID');
+      if (isPublishedCollection(json, endpointType)) {
```
Comment (arty-name) [inline src/components/ServiceEndpoints/ServiceEndpoint.tsx:78]: Do you think the flow would look nicer if we check for the happy path first and handle fallbacks later?

File: src/components/ServiceEndpoints/ServiceEndpoint.tsx [hunk 3]
```diff
@@ -1,54 +1,93 @@
...
+        setCredentials(map(json, 'credential'));
+        return;
       }
-    } catch (error) {
-      console.log(error);
-      setError('Cannot fetch the credentials from the given endpoint');
+
+      throw new Error();
```
Comment (arty-name) [inline src/components/ServiceEndpoints/ServiceEndpoint.tsx:83]: Would a message in this error help with debugging in some far ahead future?

File: src/components/ServiceEndpoints/ServiceEndpoint.tsx [hunk 4]
```diff
@@ -1,54 +1,93 @@
...
+    } catch (err) {
```
Comment (arty-name) [inline src/components/ServiceEndpoints/ServiceEndpoint.tsx:84]: Is this `err` needed?
```suggestion
    } catch {
```

**Review by arty-name — APPROVED** [review]

File: src/components/ServiceEndpoints/ServiceEndpoint.tsx (hunk 1 above)
Comment (lukeg90) [reply src/components/ServiceEndpoints/ServiceEndpoint.tsx:40]: Oh good point, we were only checking the first credential before

**Pull request merged**